jubatus/core/classifier/nearest_neighbor_classifier.cpp

```diff
@@ -260,6 +260,41 @@ void nearest_neighbor_classifier::unlearn_id(const std::string& id) {
   nearest_neighbor_engine_->get_table()->delete_row(id);
 }
 
+void nearest_neighbor_classifier::decrement_label_counter(
+    const std::string& label) {
+  util::concurrent::scoped_lock lk(label_mutex_);
+  labels_[label] -= 1;
+  if (labels_[label] <= 0) {
+    labels_.erase(label);
+  }
+}
+
+void nearest_neighbor_classifier::regenerate_label_counter() {
```

Comment:
It's better to revise names of variable names; difficult to detect typos between `labels` and `labels_`